Ported/combat-bees/Assets/Scripts/Systems/BeeRemovalSystem.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Mathematics;
+using Unity.Transforms;
+    
+[WithAll(typeof(Dead))]
+[BurstCompile]
+public partial struct BeeRemovalJob : IJobEntity
+{
+    [ReadOnly] public AABB FieldArea;
+    public EntityCommandBuffer.ParallelWriter ECB;
+    [ReadOnly] public ComponentLookup<Food> FoodComponentLookup;
+
+    [BurstCompile]
+    public void Execute([ChunkIndexInQuery] int chunkIndex, Entity entity, in LocalToWorldTransform transform, in BeeProperties beeProperties)
+    {
+        if (beeProperties.CarriedFood != Entity.Null && FoodComponentLookup.HasComponent(beeProperties.CarriedFood))
+        {
+            ECB.SetComponent(chunkIndex, beeProperties.CarriedFood, new Food{CarrierBee = Entity.Null});
+            ECB.AddComponent<UnmatchedFood>(chunkIndex, beeProperties.CarriedFood);
+        }
+
+        if (transform.Value.Position.y >= FieldArea.Min.y)
```

Comment:
@Impossible : Do you mean <=?

Reply:
Also, how are you making sure that only the bees with the component "Dead" enabled are processed here?